### PR TITLE
Fix uid conflict when duplicating resource

### PR DESCRIPTION
**Pull request by KoBeWi**
Simple fix for #54774

### CODE REVIEW

**Comment by KoBeWi** [comment]
I unlinked the issue, because the PR doesn't fix it fully. It applies to any text resource file, not sure how to handle them all. I added a fix for scenes.

**Comment by reduz** [comment]
I am checking everything and I think the best will be to add an API to change the UUID for any resource

**Comment by KoBeWi** [comment]
But the .import file fix is still relevant, no? (unless the UUID change would handle that)

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -1225,6 +1225,31 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
...
+
+			// Remove uid from .import file to avoid conflict.
+			Ref<ConfigFile> cfg;
+			cfg.instantiate();
+			cfg->load(new_path + ".import");
+			cfg->erase_section_key("remap", "uid");
+			cfg->save(new_path + ".import");
+		} else if (p_item.is_file && (old_path.get_extension() == "tscn" || old_path.get_extension() == "tres")) {
+			// FIXME: Quick hack to fix text resources. This should be fixed properly.
+			Error err;
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp:1237]:
```
editor/filesystem_dock.cpp: In member function 'void FileSystemDock::_try_duplicate_item(const FileSystemDock::FileOrFolder&, const String&) const':
editor/filesystem_dock.cpp:1237:10: error: declaration of 'err' shadows a previous local [-Werror=shadow=local]
 1237 |    Error err;
      |          ^~~
editor/filesystem_dock.cpp:1220:8: note: shadowed declaration is here
 1220 |  Error err = p_item.is_file ? da->copy(old_path, new_path) : da->copy_dir(old_path, new_path);
      |        ^~~
```
You can likely just reuse the existign one.

Comment (KoBeWi) [reply editor/filesystem_dock.cpp:1237]: Ah, I didn't notice that.

**Review by akien-mga — APPROVED** [review]
Merging this as a temporary workaround for 4.0 alpha 2. Let's keep the issue open and figure out a proper fix before beta.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**